src/simulators/statevector/qubitvector.hpp

```diff
@@ -680,17 +692,34 @@ cvector_t QubitVector<data_t>::vector() const {
 //------------------------------------------------------------------------------
 
 template <typename data_t>
-uint_t QubitVector<data_t>::index0(const reg_t& qubits_sorted, const uint_t k) const {
+template <typename list_t>
+uint_t QubitVector<data_t>::index0(const list_t &qubits_sorted, const uint_t k) const {
   uint_t lowbits, retval = k;
-  for (const auto& qubit : qubits_sorted) {
-    lowbits = retval & MASKS[qubit];
-    retval >>= qubit;
-    retval <<= qubit + 1;
+  for (size_t j = 0; j < qubits_sorted.size(); j++) {
+    lowbits = retval & MASKS[qubits_sorted[j]];
+    retval >>= qubits_sorted[j];
+    retval <<= qubits_sorted[j] + 1;
     retval |= lowbits;
   }
   return retval;
 }
 
+template <typename data_t>
+template <size_t N>
+areg_t<1ULL << N> QubitVector<data_t>::indexes(const areg_t<N> &qs,
```

Comment:
Is there a significant speed difference between this old static indexes function and the new function that uses unique pointers? It is much less convenient to use which is why it was removed.